context-monitoring/src/main/java/de/atb/context/monitoring/monitors/messagebroker/MessageBrokerMonitor.java

```diff
@@ -0,0 +1,240 @@
+        }
+    }
+
+    public void monitor() {
+        this.logger.info("Starting monitoring for MessageBroker at URI: " + this.dataSource.getUri());
+
+        this.isAnalysisRunning = true;
+        handleMessageBroker();
+        this.isAnalysisRunning = false;
+    }
+
+    protected final void handleMessageBroker() {
+        this.logger.debug("Handling URI " + this.dataSource.getUri() + "...");
+        try {
+            if ((this.dataSource.getUri() != null)) {
+                MessageBrokerAnalyser analyser = (MessageBrokerAnalyser) parser.getAnalyser();
+
+                if (parser.parse(this.dataSource.toMessageBroker())) {
+                    this.indexer.addDocumentToIndex(parser.getDocument());
+                    this.raiseParsedEvent(this.dataSource.toMessageBroker(), parser.getDocument());
+                    this.raiseAnalysedEvent(analyser.analyse(this.dataSource.toMessageBroker()),
+                            this.dataSource.toMessageBroker(), analyser.getDocument());
+                }
+            }
+            // clean up indexed document and remove all stuff already used
+            parser.getDocument().removeFields("content");
+            parser.getDocument().removeFields("monitoredAt");
+        } catch (Exception e) {
+            logger.error("Unknown error ind MessageBrokerMonitor.handleMessageBroker()", e);
+        }
+    }
+
+    /**
+     * Every X seconds this class calls the monitor method to consume all received topics from kafka and
+     * send them to the MessageBrokerParser and KafkaAnalyser
+     */
+    protected static final class MessageBrokerMonitoringRunner implements Runnable {
+        private static final Logger logger = LoggerFactory.getLogger(MessageBrokerMonitoringRunner.class);
+        private final MessageBrokerMonitor parent;
+
+        public MessageBrokerMonitoringRunner(final MessageBrokerMonitor parent) {
+            this.parent = parent;
+        }
+
+        @Override
+        public void run() {
+            try {
+                this.parent.monitor();
+            } catch (Exception e) {
+                logger.error("Error during monitoring of MessageBroker! ", e);
+            }
+        }
+
+    }
+
+    @Override
+    public void documentAnalysed(List<IMonitoringDataModel<?, ?>> analysedList, IMessageBroker parsed, Document document) {
+        if ((analysedList != null) && (analysedList.size() > 0)) {
+            for (IMonitoringDataModel<?, ?> analysed : analysedList) {
+                logger.info("Created monitoring data for " + analysed.getApplicationScenario());
+                try {
+                    this.amiRepository.persist(analysed);
+                    logger.info("Persisted monitoring data for " + analysed.getApplicationScenario());
+                } catch (ContextFault e1) {
+                    logger.error(e1.getMessage(), e1);
+                }
+            }
+        }
+        return;
```

Comment:
return is not needed in a method returning `void`

Reply:
done